pkg/sensors/program/map.go

```diff
@@ -260,12 +261,23 @@ func LoadOrCreatePinnedMap(pinPath string, mapSpec *ebpf.MapSpec) (*ebpf.Map, er
 				"map-name": mapSpec.Name,
 			}).Warn("tetragon, incompatible map found: will delete and recreate")
 			m.Close()
-			os.Remove(pinPath)
-		} else {
-			return m, nil
+			// If we are creating the map, let's ignore the compatibility error,
+			// remove the pin and create the map with our spec.
+			if create {
+				os.Remove(pinPath)
+				return createPinnedMap(pinPath, mapSpec)
+			}
+			return nil, fmt.Errorf("incompatible map path '%s'", pinPath)
 		}
+		return m, nil
+	}
+	if create {
+		return createPinnedMap(pinPath, mapSpec)
```

Comment:
Could move this create just after the state(pinPath) failure , and have the rest outside of if , just for readability up to you

Reply:
hm, so it is already in the place after the stat failure.. not sure what you mean